src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSecurityTokenHandler.cs

```diff
@@ -622,12 +652,174 @@ private IEnumerable<SecurityKey> GetAllSigningKeys(TokenValidationParameters val
         }
 
         /// <summary>
+        /// Creates claims from a <see cref="SamlAttributeStatement"/>.
+        /// </summary>
+        /// <param name="statement">The <see cref="SamlAttributeStatement"/>.</param>
+        /// <param name="identity">A <see cref="ClaimsIdentity"/>.</param>
+        /// <param name="issuer">The issuer.</param>
+        /// <exception cref="ArgumentNullException">if <see cref="SamlAttributeStatement"/> is null.</exception>
+        /// <exception cref="ArgumentNullException">if <see cref="ClaimsIdentity"/> is null.</exception>
+        /// <exception cref="SamlSecurityTokenException">if Actor existing in both identity and attributes of statement.</exception>
+        protected virtual void ProcessAttributeStatement(SamlAttributeStatement statement, ClaimsIdentity identity, string issuer)
+        {
+            if (statement == null)
+                throw LogArgumentNullException(nameof(statement));
+
+            if (identity == null)
+                throw LogArgumentNullException(nameof(identity));
+
+            foreach (var attribute in statement.Attributes)
+            {
+                if (StringComparer.Ordinal.Equals(attribute.Name, ClaimTypes.Actor))
+                {
+                    if (identity.Actor != null)
+                        throw LogExceptionMessage(new SamlSecurityTokenException(LogMessages.IDX10512));
+
+                    SetDelegateFromAttribute(attribute, identity, issuer);
+                }
+                else
+                {
+                    // each value has same issuer
+                    string originalIssuer = attribute.OriginalIssuer ?? issuer;
+                    foreach (var value in attribute.Values)
+                    {
+                        if (value != null)
+                        {
+                            var claim = new Claim(attribute.Name, value, attribute.AttributeValueXsiType, issuer, originalIssuer);
+                            if (attribute.Name != null)
```

Comment:
attribute.Name cannot be null.
We should not add these properties (ClaimProperties.SamlAttributeName, ClaimProperties.SamlAttributeNamespace) as they are just echoing the name and namespace of the claim.